partiql-tests-data/fail/static-analysis/primitives/timestamp-constructor.ion

```diff
@@ -0,0 +1,655 @@
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid Day field - minus before day",
+  statement: "TIMESTAMP '2021-03--10 00:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid Day field - plus before day",
+  statement: "TIMESTAMP '2021-03-+10 00:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid literal Format - random string",
+  statement: "TIMESTAMP 'timestamp'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid HOUR field - above upper bound",
+  statement: "TIMESTAMP '2021-03-01 24:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid HOUR field - Negative value",
+  statement: "TIMESTAMP '2021-03-01 -01:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid HOUR field - plus sign",
+  statement: "TIMESTAMP '2021-03-01 +01:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid HOUR field - plus sign",
+  statement: "TIMESTAMP '2021-03-01 +01:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid MINUTE field - above upper bound",
+  statement: "TIMESTAMP '2021-03-01 00:60:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid MINUTE field - negative value",
+  statement: "TIMESTAMP '2021-03-01 00:-01:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid MINUTE field - plus sign",
+  statement: "TIMESTAMP '2021-03-01 00:+01:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+// Setting to 62 to take account of leap second, if any implement does care.
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid SECOND field - above upper bound",
+  statement: "TIMESTAMP '2021-03-01 00:00:62'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid SECOND field - negative value",
+  statement: "TIMESTAMP '2021-03-01 00:00:-01'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid SECOND field - plus sign",
+  statement: "TIMESTAMP '2021-03-01 00:00:+01'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - SQL Fomat - invalid Precision - negative value",
+  statement: "TIMESTAMP(-1) '2021-03-01 00:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+// TIMESTAMP WITHOUT TIME ZONE - RFC 3339 FORMAT
```

Comment:
Does the SQL spec say anything about supporting the RFC 3339 format? Or is PartiQL supporting this additional timestamp format?

We could alternatively put these tests in the [partiql-tests-data-extended](https://github.com/partiql/partiql-tests/tree/main/partiql-tests-data-extended) folder since the conformance tests should just contain tests that PartiQL + SQL specify.

Reply:
In my opinion, the Ion syntax (i.e. using the backticks) is a different syntax than the one included in these tests (i.e. `TIMESTAMP` keyword followed by a string literal w/ the `T` between date and time elements) despite the contents within the backticks/quotes being the same. I think it we should move these tests to the extended directory for now. If the timestamp RFC is approved with the verbiage that PartiQL SHALL support this specific syntax (i.e. `TIMESTAMP` keyword followed by a string literal w/ the `T` between date and time elements), then we can move these tests back to the core conformance tests.

Reply:
And just so we're clear, are you equating the SHALL keyword to what [RFC2119](https://www.ietf.org/rfc/rfc2119.txt) refers to as MUST/REQUIRED/SHALL:

```
1. MUST   This word, or the terms "REQUIRED" or "SHALL", mean that the
   definition is an absolute requirement of the specification.
```

Reply:
> In my opinion, the Ion syntax (i.e. using the backticks) is a different syntax than the one included in these tests (i.e. `TIMESTAMP` keyword followed by a string literal w/ the `T` between date and time elements) despite the contents within the backticks/quotes being the same. I think it we should move these tests to the extended directory for now. If the timestamp RFC is approved with the verbiage that PartiQL SHALL support this specific syntax (i.e. `TIMESTAMP` keyword followed by a string literal w/ the `T` between date and time elements), then we can move these tests back to the core conformance tests.

I agree. Will move those. Thanks!

Reply:
> And just so we're clear, are you equating the SHALL keyword to what [RFC2119](https://www.ietf.org/rfc/rfc2119.txt) refers to as MUST/REQUIRED/SHALL:
> 
> ```
> 1. MUST   This word, or the terms "REQUIRED" or "SHALL", mean that the
>    definition is an absolute requirement of the specification.
> ```

Yes.